src/Application.php

```diff
@@ -56,4 +58,10 @@ public function addInitRoboFileCommand($roboFile, $roboClass)
         });
         $this->add($createRoboFile);
     }
+
+    protected function addSelfUpdateCommand()
+    {
+        $selfUpdateCommand = new SelfUpdateCommand('self:update', Robo::VERSION, 'consolidation/robo');
```

Comment:
Could Robo::VERSION be obtained from the Application instance?

Could 'consolidation/robo' be read from the composer.json file?

Reply:
The 'Robo as a Framework' document should be updated to describe what a derivative app needs to do to get the self:update functionality. If the above enhancements work, though, then it seems that derived apps don't need to do anything, in which case no documentation is necessary (although a note there that this is a feature would be great).

Reply:
Github Repo is not necessarily identical to the composer name .. so we would have to go to packagist and then fetch he github repo ... too much ...

Reply:
We could document the requirement that the github repo be the same as the Composer package name to use this mechanism, and also explain how to override this method for apps where this is not true.

Reply:
I don't like it ... could lead to "false positives". Github repo and composer.json are two very different things, just by convention often the same. What we could / should do is to disable the update from your main repo for apps. How can I detect it's an app?

Reply:
Well, you could read composer.json and check to see if the repository is consolidation/robo.

The better thing, I think, would be if it was possible to inject the version and repository into the Application class, and pass the correct values in from the front controller. The Application class is already provided `$version` in its constructor (you may have to cache this separately, because Symfony prefers to protect information like this in private fields); you could add the packagist project [here](https://github.com/consolidation/Robo/blob/1.1.2/robo#L19) via setter injection. Getting this into the application (which I would also do via setter injection) might be a bit inconvenient, as the Robo runtime class initialization is a bit messy (b/c support, mostly). However, the advantage here is that anyone who uses Robo as a framework must provide their own front controller, and this would be a natural place for them to provide their update project also.

Reply:
fixed